extensions/positron-python/python_files/posit/positron/convert.py

```diff
@@ -67,23 +68,25 @@ def build(self) -> List[StrictStr]:
 class CodeConverter:
     """Base class for generating dataframe code strings."""
 
-    def __init__(self, table, table_name: str, params: ConvertToCodeParams):
-        """Initialize the code generator with a default DataFrame variable name.
-
-        Parameters
-        ----------
-        table : DataFrame or Series
-            DataFrame or Series to generate code for.
-        table_name : str
-            Name of the DataFrame variable in the generated code.
-        params : ConvertToCodeParams
-            Parameters for conversion, including filters and sort keys.
-        """
+    def __init__(
+        self,
+        table,
+        table_name: str,
+        params: ConvertToCodeParams,
+        *,
+        was_series: bool = False,
+        filter_handler_class,
+        sort_handler_class,
+    ):
         self.table = table
         self.table_name: str = table_name
         self.params: ConvertToCodeParams = params
+        self.was_series = was_series
+        self.syntax_name = params.code_syntax_name.code_syntax_name
+        self.filter_handler_class = filter_handler_class
+        self.sort_handler_class = sort_handler_class
```

Comment:
These could also be class variables (default None on the base class, set to the subclass handlers in the subclasses) which would perhaps make the subclass constructors not needed